File: LinkedList/python_Implementation/Easy/2_Get_Node_Value/get_node_val.py
class Node:
    def __init__(self, val, next = None) -> None:
        self.val = val
        self.next = next

def getNodeVal(index, head, count):
    if head == None:
        return -1
    if count == index:
        return head.val
    val = getNodeVal(index, head.next, count + 1)
    return val

A = Node("A")
B = Node("B")
C = Node("C")
D = Node("D")

A.next = B
B.next = C
C.next = D

print(getNodeVal(2,A,0))
        

File: LinkedList/python_Implementation/Easy/2_Get_Node_Value/get_node_val.js
class Node{
    constructor(val, next = null){
        this.val = val;
        this.next = next;
    }
}

const getNodeVal = (index, head, count) => {
    if (head === null){
        return -1;
    }
    if (count === index){
        return head.val
    }
    let val = getNodeVal(index, head.next, count + 1);
    return val
}

const A = new Node("A");
const B = new Node("B");
const C = new Node("C");
const D = new Node("D");

A.next = B;
B.next = C;
C.next = D;

console.log(getNodeVal(2,A,0));